client/src/components/Overview/Gallery.jsx

```diff
@@ -1,21 +1,62 @@
 /* eslint-disable import/extensions */
-import React from 'react';
-import { useSelector, useDispatch } from 'react-redux';
-import { updatePhoto } from '../../reducers/Style-Reducer';
-
+import React, { useState } from 'react';
+import { useSelector } from 'react-redux';
 // eslint-disable-next-line no-unused-vars
 import overviewStyling from './overview.css';
 import GalleryThumbnail from './GalleryThumbnail.jsx';
 
 const Gallery = () => {
-  const dispatch = useDispatch();
-
   const selectedStyle = useSelector((state) => state.style.style);
   const stylePhotos = useSelector((state) => state.style.photos);
-  const mainImage = useSelector((state) => state.style.mainPhoto);
+  const [mainPhotoIndex, setPhotoIndex] = useState(0);
+
+  const mainImage = stylePhotos[mainPhotoIndex];
+
+  // sets the main image based on index o thumbnail selected
+  const selectPhoto = (index) => {
+    setPhotoIndex(index);
+  };
+
+  const moveUp = () => {
+    if (mainPhotoIndex > 0) {
+      setPhotoIndex(mainPhotoIndex - 1);
+    }
+  };
 
-  const selectPhoto = (photo) => {
-    dispatch(updatePhoto(photo));
+  const moveDown = () => {
+    if (mainPhotoIndex < stylePhotos.length - 1) {
+      setPhotoIndex(mainPhotoIndex + 1);
+    }
+  };
+
+  const handleKeyDown = (e) => {
+    if (e.keyCode === 37 || e.keyCode === 38) { // left or up
+      moveUp();
+    } else if (e.keyCode === 39 || e.keyCode === 40) { // right or down
+      moveDown();
+    }
+  };
+
+  const renderUpArrow = () => {
+    if (stylePhotos.length > 7 && mainPhotoIndex > 0) {
+      return (
+        <span role="button" tabIndex="-1" onClick={moveUp} onKeyDown={handleKeyDown}>
+          <img className="thumbnail-arrow" alt="up arrow" src="https://img.icons8.com/external-those-icons-fill-those-icons/24/ffffff/external-up-arrows-those-icons-fill-those-icons.png" />
+        </span>
+      );
+    }
+    return <span />;
+  };
+
+  const renderDownArrow = () => {
+    if (stylePhotos.length > 7 && mainPhotoIndex < stylePhotos.length - 1) {
+      return (
+        <span role="button" tabIndex="-1" onClick={moveDown} onKeyDown={handleKeyDown}>
+          <img className="thumbnail-arrow" alt="down arrow" src="https://img.icons8.com/external-those-icons-fill-those-icons/24/ffffff/external-down-arrows-those-icons-fill-those-icons-1.png" />
+        </span>
+      );
+    }
+    return <span />;
```

Comment:
Same as above, return null rather than empty span

Reply:
@afong888 Same here. I'll update to null. Thanks!